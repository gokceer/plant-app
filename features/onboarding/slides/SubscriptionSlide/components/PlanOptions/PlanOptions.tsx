import { useState } from "react";
import { StyleSheet, View } from "react-native";
import PlanCard from "../PlanCard/PlanCard";

interface Plan {
  id: "monthly" | "yearly";
  title: string;
  description: string;
  badge?: string;
}

const PLANS: Plan[] = [
  {
    id: "monthly",
    title: "1 Month",
    description: "$2.99/month, auto renewable",
  },
  {
    id: "yearly",
    title: "1 Year",
    description: "First 3 days free, then $529.99/year",
    badge: "Save 50%",
  },
] as const;

type PlanId = Plan["id"];

const PlanOptions = () => {
  const [selectedPlan, setSelectedPlan] = useState<PlanId>("yearly");

  return (
    <View style={styles.planContainer}>
      {PLANS.map((plan) => (
        <PlanCard
          testID={`planCard${plan.id}`}
          key={plan.id}
          title={plan.title}
          description={plan.description}
          badge={plan?.badge}
          selected={selectedPlan === plan.id}
          onPress={() => setSelectedPlan(plan.id)}
        />
      ))}
    </View>
  );
};

export default PlanOptions;

const styles = StyleSheet.create({
  planContainer: {
    marginTop: 20,
  },
});
