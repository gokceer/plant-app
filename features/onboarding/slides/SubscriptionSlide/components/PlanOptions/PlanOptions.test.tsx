import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import PlanOptions from "./PlanOptions";

describe("PlanOptions", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<PlanOptions />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("initially selects the yearly plan", () => {
    const { getByTestId } = render(<PlanOptions />);

    const yearlyCard = getByTestId("planCardyearly");
    const monthlyCard = getByTestId("planCardmonthly");

    expect(yearlyCard.props?.accessibilityState?.selected).toBe(true);
    expect(monthlyCard.props?.accessibilityState?.selected).toBe(false);
  });

  it("changes selection when a plan card is pressed", () => {
    const { getByTestId } = render(<PlanOptions />);

    const monthlyCard = getByTestId("planCardmonthly");
    const yearlyCard = getByTestId("planCardyearly");

    expect(yearlyCard.props?.accessibilityState?.selected).toBe(true);
    expect(monthlyCard.props?.accessibilityState?.selected).toBe(false);

    fireEvent.press(monthlyCard);

    expect(monthlyCard.props?.accessibilityState?.selected).toBe(true);
    expect(yearlyCard.props?.accessibilityState?.selected).toBe(false);
  });
});
