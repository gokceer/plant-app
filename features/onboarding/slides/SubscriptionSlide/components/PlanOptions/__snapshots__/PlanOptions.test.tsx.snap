// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanOptions renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "marginTop": 20,
    }
  }
>
  <View
    accessibilityRole="radio"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 14,
          "marginBottom": 16,
          "padding": 16,
        },
        {
          "backgroundColor": "#FFFFFF0D",
        },
      ]
    }
    testID="planCardmonthly"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF26",
              "borderRadius": 12,
              "height": 24,
              "justifyContent": "center",
              "width": 24,
            },
            false,
          ]
        }
        testID="radioOuterView"
      />
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "position": "relative",
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontFamily": "RubikMedium",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            1 Month
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFFB2",
              "fontFamily": "RubikLight",
              "fontSize": 14,
              "marginTop": 4,
            }
          }
        >
          $2.99/month, auto renewable
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityRole="radio"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": true,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 14,
          "marginBottom": 16,
          "padding": 16,
        },
        {
          "borderColor": "#28AF6E",
          "borderWidth": 1.5,
        },
      ]
    }
    testID="planCardyearly"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF26",
              "borderRadius": 12,
              "height": 24,
              "justifyContent": "center",
              "width": 24,
            },
            {
              "backgroundColor": "#28AF6E",
            },
          ]
        }
        testID="radioOuterView"
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 4,
              "height": 8,
              "width": 8,
            }
          }
          testID="radioInnerView"
        />
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "position": "relative",
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontFamily": "RubikMedium",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            1 Year
          </Text>
          <View
            style={
              {
                "backgroundColor": "#34C759",
                "borderBottomLeftRadius": 20,
                "borderTopRightRadius": 12,
                "paddingHorizontal": 12,
                "paddingVertical": 4,
                "position": "absolute",
                "right": -16,
                "top": -16,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "RubikMedium",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              Save 50%
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFFB2",
              "fontFamily": "RubikLight",
              "fontSize": 14,
              "marginTop": 4,
            }
          }
        >
          First 3 days free, then $529.99/year
        </Text>
      </View>
    </View>
  </View>
</View>
`;
